src/olympia/versions/tests/test_update.py

```diff
@@ -0,0 +1,358 @@
+        )
+        # The newer version is not approved, so it won't be picked up.
+        self._check_ok(expected_version)
+
+    def test_newer_version_disabled(self):
+        expected_version = self.addon.current_version
+        version_factory(
+            addon=self.addon,
+            min_app_version='58.0',
+            max_app_version='*',
+            version='1.1',
+            file_kw={'status': amo.STATUS_DISABLED, 'hash': 'fakehash11'},
+        )
+        # The newer version is disabled, so it won't be picked up.
+        self._check_ok(expected_version)
+
+    def test_no_unlisted_version(self):
+        expected_version = self.addon.current_version
+        version_factory(
+            addon=self.addon,
+            min_app_version='58.0',
+            max_app_version='*',
+            version='1.1',
+            channel=amo.RELEASE_CHANNEL_UNLISTED,
+            file_kw={'hash': 'fakehash1.1'},
+        )
+        # The newer version is unlisted, so it won't be picked up.
+        self._check_ok(expected_version)
+
+    def test_no_deleted_version(self):
+        expected_version = self.addon.current_version
+        version = version_factory(
+            addon=self.addon,
+            min_app_version='58.0',
+            max_app_version='*',
+            version='1.1',
+            file_kw={'hash': 'fakehash1.1'},
+        )
+        version.delete()
+        # The newer version is deleted, so it won't be picked up.
+        self._check_ok(expected_version)
+
+    def test_different_addon(self):
+        expected_version = self.addon.current_version
+        addon_factory(
+            version_kw={
+                'min_app_version': '57.0',
+                'max_app_version': '*',
+                'version': '1.1',
+            },
+            file_kw={'hash': 'fakehash1.1'},
+        )
+        # The newer version is for a different add-on so it won't be picked up.
+        self._check_ok(expected_version)
+
+    def test_no_compat_addon(self):
+        # No updates will be found for add-ons with no compat in default
+        # (strict) mode.
+        self.addon.current_version.apps.all().delete()
+        self._check_no_updates()
+
+    def test_ignore_mode(self):
+        self.query_params['compatMode'] = 'ignore'
+        self.test_basic()
+
+    def test_ignore_mode_strict_compatibility_version(self):
+        self.query_params['compatMode'] = 'ignore'
+        expected_version = version_factory(
+            addon=self.addon,
+            min_app_version='58.0',
+            max_app_version='58.*',
+            version='1.1',
+            file_kw={'strict_compatibility': True, 'hash': 'fakehash11'},
+        )
+        # Despite having strict compatibility set to 58.* max, the newer
+        # version _will_ get picked up as we're in ignore mode.
+        self._check_ok(expected_version)
+
+    @pytest.mark.xfail(reason='Need to sort out compatibility info for dictionaries')
```

Comment:
if we're intending to fix this can we have a follow-up issue?  If not may as well delete the test